app/services/shelf_list.rb

```diff
@@ -35,6 +35,14 @@ def build
     }
   end
 
+  def forward_shelfkey
+    "forward_#{classification}_shelfkey"
+  end
+
+  def reverse_shelfkey
+    "reverse_#{classification}_shelfkey"
+  end
```

Comment:
I had originally intended for the shelfkey fields to be passed in during initialization so that this class didn't need to know about how they were named, but I completely understand why you need this now.